lib/rubygems/installer.rb

```diff
@@ -698,24 +698,23 @@ def process_options # :nodoc:
 
   def check_that_user_bin_dir_is_in_path # :nodoc:
     user_bin_dir = @bin_dir || Gem.bindir(gem_home)
-    user_bin_dir = user_bin_dir.gsub(File::SEPARATOR, File::ALT_SEPARATOR) if
-      File::ALT_SEPARATOR
 
     path = ENV['PATH']
+
+    # Windows paths are case insensitive, but with either ALT_SEPARATOR
```

Comment:
HFS is case insensitive, too. Is that relevant here?

Reply:
How can one test for HFS?

Reply:
You can test for case insensitivity in general with something like this:

```
require 'tempfile'
tempfile = Tempfile.new
p File.exists?(tempfile.path.downcase) && File.exists?(tempfile.path.upcase)
```
On my Linux laptop, this prints false. On my Mac, this prints true.

Reply:
Rather than write a file, how about
```ruby
if __FILE__ != __FILE__.downcase
  FS_CASE_INSENS = File.exists?(__FILE__.downcase)
elsif __FILE__ != __FILE.upcase
  FS_CASE_INSENS = File.exists?(__FILE__.upcase)
else
  puts "indeterminate?"
end
p "FS_CASE_INSENS #{FS_CASE_INSENS}"
```

Reply:
@colby-swandale

Re the above issue, I know there is similar code elsewhere in RG (I may have done a PR?).  Not sure if it was tests or 'code'.

Might something like this be added as a constant?  I may suggest it in ruby/ruby, as I only thought it was windows...

Reply:
@MSP-Greg Ah, true, we could just use `__FILE__`, good idea!

Reply:
@djberg96 - updated revised code...

Reply:
@MSP-Greg sorry, i didn't see this comment. A constant sounds great!

Reply:
@colby-swandale

The issue of determining whether the location in question is case insensitive was also discussed in Ruby core, and Nobu had some good suggestions which are included in the current PR (there's more than a few updates).

There are some edge cases where things get messy, like a path where `path == path.swapcase`, networked drives/symlinks, etc.  I believe the code defaults to 'case sensitive` in these cases.